### ISSUE TITLE
Cannot build Release

### ISSUE THREAD

**Issue by guiguem** [project8/katydid#139]
When trying to build Katydid in Release mode on my mac, I get:
```
make
[ 10%] Built target yaml-cpp
[ 16%] Built target Scarab_Nymph_Katydid
[ 16%] Building CXX object Nymph/Library/CMakeFiles/Nymph_Katydid.dir/Application/KTProcessorToolbox.cc.o
In file included from /Users/guig098/Work/Project8/SourceTree/katydid/Nymph/Library/Application/KTProcessorToolbox.cc:8:
In file included from /Users/guig098/Work/Project8/SourceTree/katydid/Nymph/Library/Application/KTProcessorToolbox.hh:12:
In file included from /Users/guig098/Work/Project8/SourceTree/katydid/Nymph/Library/Utility/KTConfigurable.hh:11:
In file included from /Users/guig098/Work/Project8/SourceTree/katydid/Nymph/Scarab/library/param/param.hh:13:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/map:442:
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/__tree:1819:22: error: the specified comparator type does not provide a const call operator [-Werror,-Wuser-defined-warnings]
                     __trigger_diagnostics()), "");
                     ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:400:28: note: in instantiation of member function 'std::__1::__tree<Nymph::KTProcessorToolbox::Thread,
      Nymph::KTProcessorToolbox::CompareThread, std::__1::allocator<Nymph::KTProcessorToolbox::Thread> >::~__tree' requested here
class _LIBCPP_TEMPLATE_VIS set
                           ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/__tree:970:7: note: from 'diagnose_if' attribute on '__trigger_diagnostics':
      _LIBCPP_DIAGNOSE_WARNING(!__invokable<_Compare const&, _Tp const&, _Tp const&>::value,
      ^                        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/__config:1095:20: note: expanded from macro '_LIBCPP_DIAGNOSE_WARNING'
    __attribute__((diagnose_if(__VA_ARGS__, "warning")))
                   ^           ~~~~~~~~~~~
1 error generated.
make[2]: *** [Nymph/Library/CMakeFiles/Nymph_Katydid.dir/Application/KTProcessorToolbox.cc.o] Error 1
make[1]: *** [Nymph/Library/CMakeFiles/Nymph_Katydid.dir/all] Error 2
make: *** [all] Error 2
```
This issue appears in debug mode too, but these would just be warnings.

I am using
```
gcc --version
Configured with: --prefix=/Applications/Xcode.app/Contents/Developer/usr --with-gxx-include-dir=/usr/include/c++/4.2.1
Apple LLVM version 9.1.0 (clang-902.0.39.1)
Target: x86_64-apple-darwin17.5.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
``` 
installed from Brew.

**Comment by guiguem** [comment]
Similar warnings are being produced in the travis build: should try to see if the same happens in Release mode.

**Comment by nsoblath** [comment]
This was a Nymph bug fixed in v1.4.5
